core/src/main/java/io/kestra/core/schedulers/AbstractScheduler.java

```diff
@@ -183,26 +220,53 @@ public void run() {
     // must be synchronized as it update schedulableNextDate and schedulable, and will be executed on the flow listener thread
     private synchronized void computeSchedulable(List<Flow> flows) {
         this.schedulableNextDate = new HashMap<>();
-
+        if (!this.started) {
+            triggerState.findAllForAllTenants().forEach(trigger -> {
+                triggerStateSaved.put(trigger.uid(), trigger);
+            });
+        }
+        this.started = true;
         this.schedulable = flows
             .stream()
             .filter(flow -> flow.getTriggers() != null && !flow.getTriggers().isEmpty())
             .filter(flow -> !flow.isDisabled() && !(flow instanceof FlowWithException))
             .flatMap(flow -> flow.getTriggers()
                 .stream()
                 .filter(abstractTrigger -> !abstractTrigger.isDisabled() && abstractTrigger instanceof PollingTriggerInterface)
-                .map(trigger -> {
-                    RunContext runContext = runContextFactory.of(flow, trigger);
-
-                    return new FlowWithTrigger(
+                .map(abstractTrigger -> {
+                    RunContext runContext = runContextFactory.of(flow, abstractTrigger);
+                    ConditionContext conditionContext = conditionService.conditionContext(runContext, flow, null);
+                    Trigger triggerContext = null;
+                    try {
+                        Trigger lastTriggers = triggerStateSaved.get(Trigger.uid(flow, abstractTrigger));
+                        if (lastTriggers != null && lastTriggers.getNextExecutionDate() == null) {
```

Comment:
Is it needed to compute the next evaluation date here? On which scenario would it be null?

Reply:
I have improved it in a new commit, but it can be null when going from a previous version of Kestra where all last triggers won't have any data yet.

The property is needed as the next object built on top of the schedulable is based on this object and requires it.

Reply:
> it can be null when going from a previous version of Kestra where all last triggers won't have any data yet.

This needs to be documented! Please add a comment